utilix/batchq.py

```diff
@@ -1,265 +1,300 @@
+        if not os.path.exists(image):
+            raise FileNotFoundError(f"Singularity image {image} does not exist")
+        new_job_string = (
+            f"unset X509_CERT_DIR\n"
+            f'if [ "$INSTALL_CUTAX" == "1" ]; then unset CUTAX_LOCATION; fi\n'
+            f"module load singularity\n"
+            f"singularity exec {bind_string} {image} {exec_file}\n"
+            f"exit_code=$?\n"
+            f"rm {exec_file}\n"
+            f"if [ $exit_code -ne 0 ]; then\n"
+            f"    echo Python script failed with exit code $exit_code\n"
+            f"    exit $exit_code\n"
+            f"fi\n"
+        )
+        if file_discriptor is not None:
+            os.close(file_discriptor)
+        return new_job_string
+
+    def submit(self):
+        """
+        Submit the job to the SLURM queue.
+        """
+        # Initialize a dictionary with mandatory parameters
+        slurm_kwargs = {
+            "job_name": self.jobname,
+            "output": self.log,
+            "qos": self.qos,
+            "error": self.log,
+            "account": self.account,
+            "partition": self.partition,
+            "mem_per_cpu": self.mem_per_cpu,
+            "cpus_per_task": self.cpus_per_task,
+        }
+
+        # Conditionally add optional parameters if they are not None
+        if self.hours is not None:
+            slurm_kwargs["time"] = datetime.timedelta(hours=self.hours)
+        if self.node is not None:
+            slurm_kwargs["nodelist"] = self.node
+        if self.exclude_nodes is not None:
+            slurm_kwargs["exclude"] = self.exclude_nodes
+        if self.dependency is not None:
+            slurm_kwargs["dependency"] = {"afterok": self.dependency}
+            slurm_kwargs["kill_on_invalid"] = "yes"
+
+        # Create the Slurm instance with the conditional arguments
+        slurm = Slurm(**slurm_kwargs)
+
+        # Process the jobstring with the container if specified
+        if self.container is not None:
+            self.jobstring = self.__singularity_wrap()
+        elif self.verbose:
+            print(f"No container specified, running job as is")
+
+        # Add the job command
+        slurm.add_cmd(self.jobstring)
+
+        print(f"Your log is located at: {self.log}")
+
+        # Handle dry run scenario
+        if self.verbose or self.dry_run:
+            print(f"Generated slurm script:\n{slurm.script()}")
+
+        if self.dry_run:
+            return
+        # Submit the job
+        slurm.sbatch()
+
+
+def submit_job(*args, **kwargs):
+    """
+    Adapter to old function name.
+    You should use JobSubmission to get modern code editor support.
 
+    Args:
+        **kwargs: Keyword arguments to pass to JobSubmission
+    """
+    logger.warning(
+        "Using legacy function name, please use JobSubmission to get modern code editor support"
+    )
+    job = JobSubmission(*args, **kwargs)
+    job.submit()
+
+
+def count_jobs() -> int:
```

Comment:
The argument for `count_jobs()` is changed here. In order to avoid conflict with the previous codes using batchq, please keep consistent with the original function